tokio-trace/src/lib.rs

```diff
@@ -61,14 +61,22 @@ macro_rules! callsite {
         impl callsite::Callsite for MyCallsite {
             fn add_interest(&self, interest: Interest) {
                 let current_interest = self.interest();
-                if interest > current_interest {
-                    let interest = match interest {
-                        Interest::NEVER => 0,
-                        Interest::ALWAYS => 2,
-                        _ => 1,
-                    };
-                    INTEREST.store(interest, Ordering::Relaxed);
-                }
+                let interest = match () {
```

Comment:
This code is admittedly more cumbersome (and potentially, flakier) than just having a defined ordering...

Reply:
I don't disagree, but I would like to be paranoid initially. We can always add in the future, but not remove.

Reply:
Specifically, I can imagine a scenario where order is no longer possible. For example, if `Interest` starts tracking specific fields.

Reply:
👍